src/system/AdminModule/Controller/AdminInterfaceController.php

```diff
@@ -0,0 +1,469 @@
+                $update_show = false;
+            }
+            
+            foreach ($newVersionInfo as $version) {
+                if (! is_array($version)) {
+                    // Invalid response, probably api limits encountered.
+                    $update_show = false;
+                }
+                if (! array_key_exists('prerelease', $version) || $version['prerelease']) {
+                    continue;
+                }
+                if (array_key_exists('tag_name', $version)) {
+                    if (version_compare($version['tag_name'], $onlineVersion) == 1) {
+                        $onlineVersion = $version['tag_name'];
+                    }
+                }
+            }
+            
+            if ($onlineVersion == '') {
+                $update_show = false;
+            }
+            $this->get('zikula_extensions_module.api.variable')->set('ZConfig', 'updateversion', $onlineVersion);
+        }
+        
+        // compare with db Version_Num
+        // if 1 then there is a later version available
+        if (version_compare($onlineVersion, $this->get('zikula_extensions_module.api.variable')->get('ZConfig', 'Version_Num') == 1)) {
+            $update_show = true;
+            $update_version = $onlineVersion;
+        } else {
+            $update_show = false;
+        }
+        
+        return $this->render("ZikulaAdminModule:AdminInterface:updatecheck.html.twig", [
+            'update_show' => $update_show,
+            'update_version' => $update_version
+        ]);
+    }
+
+    /**
+     * @Route("/menu")
+     *
+     * Admin menu.
+     *
+     * @param string $mode
+     *            string $template
+     *            
+     * @return Response symfony response object
+     */
+    public function menuAction()
+    {
+        if (! $this->hasPermission('ZikulaAdminModule::', '::', ACCESS_ADMIN)) {
+            throw new AccessDeniedException();
+        }
+        
+        $masterRequest = $this->get('request_stack')->getMasterRequest();
+        $currentRequest = $this->get('request_stack')->getCurrentRequest();
+        // get caller info
+        $caller = [];
+        $caller['_zkModule'] = $masterRequest->attributes->get('_zkModule');
+        $caller['_zkType'] = $masterRequest->attributes->get('_zkType');
+        $caller['_zkFunc'] = $masterRequest->attributes->get('_zkFunc');
+        $caller['path'] = $masterRequest->getPathInfo();
+        $caller['info'] = \ModUtil::getInfoFromName($caller['_zkModule']);
+        // category we are in
+        $requestedCid = $masterRequest->attributes->get('acid');
+        if ($caller['_zkModule'] == 'ZikulaAdminModule' || $caller['_zkModule'] == '') {
+            $cid = empty($requestedCid) ? $this->getVar('startcategory') : $requestedCid;
+        } else {
+            $cid = \ModUtil::apiFunc('ZikulaAdminModule', 'admin', 'getmodcategory', array(
```

Comment:
This may be a good time to write an Api for AdminModule to replace these ModUtil calls. (also below). See other new Api classes. It would need an injection of the Repository class and type hinted by an Interface (for example see VariableApi).